agent-inject/agent/agent.go

```diff
@@ -577,15 +590,34 @@ func (a *Agent) Validate() error {
 }
 
 func serviceaccount(pod *corev1.Pod) (string, string) {
-	var serviceAccountName, serviceAccountPath string
+
```

Comment:
Should this just return a ServiceAccountTokenVolume type?

I'd also like to see some unit tests for this function since the logic is getting a little complex.